### PR TITLE
Split server tests into multiple test files

### PR DESCRIPTION
**Pull request by manuelwedler**
See #1357 

The biggest improvement of this PR are the `ServerFixture` and `LocalChainFixture` that allow to spin up a server and a local chain for testing. They can be created inside a `describe` block in order to not have one shared server for all tests. In the future we can tweak the options of the fixtures to provide configurations as they are required by specific tests.

Other than that, I created two separate folders inside `test`. One for unit and one for integration tests, since there were also a few unit tests in the server.js file. Overall, this should provide the right structure to keep the tests maintainable in the future when we add more tests.

### CODE REVIEW

File: services/server/package.json [hunk 1]
```diff
@@ -14,9 +14,14 @@
...
     "postgres-test:stop": "docker-compose -f test/docker-compose.yml rm -fsv",
     "postgres-test:migrate": "cd ../database && npm run migrate:up -- --env test",
-    "test": "npm run postgres-test:migrate && NODE_CONFIG_DIR=./dist/config DOTENV_CONFIG_PATH=./.env c8 --reporter=none mocha -r dotenv/config --exit",
-    "test-local": "export DOCKER_HOST_POSTGRES_TEST_PORT=${DOCKER_HOST_POSTGRES_TEST_PORT:-5431} && npm run postgres-test:start && echo \"Sleeping 5 seconds to wait for the container to start\" && sleep 5 && npm run postgres-test:migrate && NODE_CONFIG_DIR=./dist/config DOTENV_CONFIG_PATH=./.env c8 --reporter=none mocha -r dotenv/config --exit; status=$?; npm run postgres-test:stop; exit $status",
-    "test:chains": "mocha test/chains/chain-tests.spec.ts --reporter mochawesome --reporter-options reportDir=chain-tests-report,reportFilename=report --exit",
+    "test:unit": "c8 --reporter=none mocha  --exit test/unit/**",
+    "test:integration": "npm run postgres-test:migrate && c8 --reporter=none mocha --exit test/integration/**",
+    "test": "npm run test:unit && npm run test:integration",
```
Comment (kuzdogan) [inline services/server/package.json:19]: Why did we separate it as :unit and :integration? Just for a way to separate tests or for some technical reason?

**Review by kuzdogan — CHANGES_REQUESTED** [review]
Small thing and a question. Otherwise LGTM

File: services/server/test/integration/verification-handlers/verify.session.spec.ts [hunk 2]
```diff
@@ -0,0 +1,839 @@
...
+      .request(serverFixture.server.app)
+      .post("/session/input-files")
+      .attach("files", zippedTruffleBuffer)
+      .then((res) => {
+        chai.expect(res.status).to.equal(StatusCodes.OK);
+        chai.expect(res.body.contracts).to.have.lengthOf(3);
+        done();
+      });
+    it("should correctly handle when uploaded 0/2 and then 1/2 sources", (done) => {
+      const metadataPath = path.join(
```
[the remark this answers is not in the document]
Comment (manuelwedler) [reply services/server/test/integration/verification-handlers/verify.session.spec.ts:479]: Looks like it was like this before. But should be changed of course

[the remark this answers is not in the document]
Comment (manuelwedler) [reply services/server/test/integration/verification-handlers/verify.session.spec.ts]: These two tests were actually never working correctly... just fixed them

**Review by kuzdogan — APPROVED** [review]

**Pull request merged**